tests/functional/test_charm.py

```diff
@@ -0,0 +1,185 @@
+import logging
+import re
+
+import pytest
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+async def test_build_and_deploy(ops_test, series):
+    """Test building apt-mirror charm and deploying it with a bundle file."""
+    charm = await ops_test.build_charm(".")
+    assert charm, "Charm was not built successfully."
+
+    await ops_test.model.deploy(
+        ops_test.render_bundle(
+            "tests/functional/bundle.yaml.j2",
+            charm=charm,
+            series=series,
+        )
+    )
+    await ops_test.model.wait_for_idle(apps=["apt-mirror"], status="blocked")
+    await ops_test.model.wait_for_idle(apps=["nginx"], status="active")
+
+    app = ops_test.model.applications["apt-mirror"]
+    status_msg = app.units[0].workload_status_message
+    assert bool(
+        re.search("^Last sync: .* not published$", status_msg)
+    ), "apt-mirror did not show correct block message."
+
+
+@pytest.mark.usefixtures("apt_mirror_unit", "configs", "helper")
```

Comment:
One more think, we specify these  these fixtures `"apt_mirror_unit", "configs", "helper"`  to be passed, but "ops_test" is also passed to each function. Why is that? Because `ops_test` is used in those fixtures? Shouldn't we specify this directly?

Reply:
After reading the documentation again, I found out that I mis-used `pytest.mark.usefixtures` here, so I removed it. `ops_test` is used to access the model --> check and wait for the status of an application, so some of the tests will also request `ops_test`.

Reply:
Thank you for looking at it.